keps/sig-node/5419-pod-level-resources-in-place-resize/README.md

```diff
@@ -0,0 +1,908 @@
+following core design principles:
+
+1. Consistency with Container-Level IPPR: The mechanism for actuating pod-level
+   resizes should be as consistent as possible with the existing container-level
+   IPPR to minimize new complexities.
+
+2. Observability: Users and cluster components must have clear visibility into the
+   actual, allocated pod-level resources post-resize.
+
+### Components/Features changes
+
+#### PodStatus API changes
+Extend `PodStatus` to include pod-level analog of the container status resource
+fields. Pod-level resource information in `PodStatus` is essential for pod-level
+[In-Place Pod Update] (https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/1287-in-place-update-pod-resources/README.md#api-changes)
+as it provides a way to track, report and use the actual resource allocation for the
+pod, both before and after a resize operation.
+
+```
+type PodStatus struct {
+...
+  // Resources represents the compute resource requests and limits that have been
+  // applied at the pod level. If pod-level resources are not explicitly specified,
+  // then these will be the aggregate resources computed from containers. If limits are 
+  // not defined for all containers (and pod-level limits are also not set), those
+  // containers remain unrestricted, and no aggregate pod-level limits will be applied.  
+  // Pod-level limit aggregation is only performed, and is meaningful only, when all 
+  // containers have defined limits.
+  // +featureGate=InPlacePodVerticalScaling
+  // +featureGate=PodLevelResources
+  // +featureGate=InPlacePodLevelResourcesVerticalScaling
+  // +optional
+  Resources *ResourceRequirements
+
+  // AllocatedResources is the total requests allocated for this pod by the node.
+  // Kubelet sets this to the accepted requests when a pod (or resize) is admitted.
+  // If pod-level requests are not set, this will be the total requests aggregated
+  // across containers in the pod.
+  // +featureGate=InPlacePodVerticalScaling
+  // +featureGate=PodLevelResources
+  // +featureGate=InPlacePodLevelResourcesVerticalScaling
+  // +optional
+  AllocatedResources ResourceList
+}
+```
+
+##### Resize Restart Policy
+
+Pod-level resize policy is not supported in the alpha stage of Pod-level resource
+feature. While a pod-level resize policy might be beneficial for VM-based runtimes
+like Kata Containers (potentially allowing the hypervisor to restart the entire VM
+on resize), this is a topic for future consideration. We plan to engage with the
+Kata community to discuss this further and will re-evaluate the need for a pod-level
+policy in subsequent development stages.
+
+The absence of a pod-level resize policy means that container restarts are
+exclusively managed by their individual `resizePolicy` configs. And, if the
+containers do not specify `resizePolicy`, `PreferNoRestart` is the default resize
+policy. The resizes (pod-level or container-level) will not result in container restarts.
```

Comment:
nit: The IPPR KEP has a note that `PreferNoRestart` restart policy for resize does not guarantee that a container won't be restarted. I assume the same is true here?

Reply:
good catch! I missed reading the that note from the IPPR KEP. Removed the statement: The resizes (pod-level or container-level) "will" not result in container restarts.
and added the same note here as well.